src/Microsoft.Restier.Core/Query/DefaultQueryHandler.cs

```diff
@@ -270,8 +270,17 @@ private Expression Process(Expression visited, Expression processed)
                     {
                         if (!visited.Type.IsAssignableFrom(filtered.Type))
                         {
-                            throw new InvalidOperationException(
-                                Resources.ProcessorCannotChangeExpressionType);
+                            // In order to filter on the navigation properties, the type is changed from ICollection<> to IQueryable<>
+                            var collectionType = visited.Type.FindGenericType(typeof(ICollection<>));
+                            var queryableType = filtered.Type.FindGenericType(typeof(IQueryable<>));
+                            if (collectionType == null 
+                                || queryableType == null 
+                                || (collectionType.GenericTypeArguments[0] != queryableType.GenericTypeArguments[0] 
```

Comment:
extract `collectionType.GenericTypeArguments[0]` and `queryableType.GenericTypeArguments[0]` to local vars to make them clearer.